src/tutorial/output.md

```diff
@@ -1,15 +1,136 @@
 # Output
 
+## Printing "Hello World"
+
+```rust
+println!("Hello World");
+```
+
+Well, that was easy.
+Great, onto the next topic.
+
+## Using println
+
+You can pretty much print all the things you like
+with the `println!` macro.
+This macro has some pretty amazing capabilities,
+but also a special syntax.
+It expects you to write,
+as a first parameter,
+a literal string that with placeholders,
+followed by the data you want to put into these placeholders
+as further arguments.
+
+For example:
+
+```rust
+let x = 42;
+println!("My lucky number is {}.", x);
+```
+
+will print
+
+> My lucky number is 42.
+
+The curly braces (`{}`) in the string above is one of these placeholders.
+This is the default placeholder type
+that tries to print the given in a human readable way.
+For numbers and strings this works very well,
+but not all types can do that.
+This is why there is also a "debug representation",
+that you can get by filling the braces of the placeholder like this: `{:?}`.
+
+For example,
+
+```rust
+let xs = vec![1, 2, 3];
+println!("The list is: {:?}", xs);
+```
+
+will print
+
+> The list is: [1, 2, 3]
+
+If you want your own data types to be printable for debugging and logging,
+you can in most cases add a `#[derive(Debug)]` above their definition.
+
+<aside>
+
+**Aside:**
+The human readable printing is done using the [`Display`] trait,
+debug output uses the [`Debug`] trait.
+You can find more information about the syntax you can use in `println!`
+in the [documentation for the `std::fmt` module][std::fmt].
+
+[`Display`]: https://doc.rust-lang.org/1.27.2/std/fmt/trait.Display.html
+[`Debug`]: https://doc.rust-lang.org/1.27.2/std/fmt/trait.Debug.html
+[std::fmt]: https://doc.rust-lang.org/1.27.2/std/fmt/index.html
+
+</aside>
+
+## Printing errors
+
+<aside class="todo">
+
+**TODO:**
+`eprintln!` and stdout vs stderr
+
+</aside>
+
+## Showing a progress bar
+
+Some CLI applications run less than a second,
+others take minutes or hours.
+If you are writing one of the latter types of programs,
+you might want to show the user that something is happening.
+For this, you should try to printing useful status updates,
+ideally in a form that can be easily consumed.
+
+Using the [indicatif] crate,
```

Comment:
Do we only want to mention one crate here or maybe also link to https://github.com/FGRibreau/spinners (it has a `shark` spinner! :shark: )

Reply:
I wanted to expand this with an example that uses indicatif to show progress as well as log messages. The spinners crate can't do that afaik